### ISSUE TITLE
The list of suggestion

### ISSUE THREAD

**Issue by arbdevml** [marcopeocchi/yt-dlp-web-ui#62]
Hello. First of all very big thank you for this project.
What do you think about these:
1) Add a date picker to download videos from the channel by specific date range,
--datebefore DATE               Download only videos uploaded on or before
                                this date. The date formats accepted is the
                                same as --date
--dateafter DATE                Download only videos uploaded on or after
                                this date.
                                
2) Add embed-thumbnail to the video file,
--embed-thumbnail     Embed thumbnail in the video as cover art.

3) Write comment of the video,
--write-comments

4) Add comment viewer,
https://gist.github.com/pukkandan/ee737fec64822f2552caf3ca4cbf5db7

5) Write a video description of the video,
 --write-description   Write video description to a .description file.
 
 6) Download subtitles of the video,
 Select the language and download the srt file of the subtitle.
 
 7) Download the channel or playlist.
 
 8) Automatically download more the new videos from channel or playlist,
 (Can be implemented using CRON and API request).
 
 9) Add Nginx reverse proxy config example.
 
 10) Add example integration with Authelia.
 
 11) Create a download template. Save the download argument to reuse.
 
 12) Ability to download mp4, add drop-down list with video formats,
480p
bestvideo[height<=480][ext=mp4]+bestaudio[ext=m4a]/best[height<=480][ext=mp4]
720p
bestvideo[height<=720][ext=mp4]+bestaudio[ext=m4a]/best[height<=720][ext=mp4]
1080p
bestvideo[height<=1080][ext=mp4]+bestaudio[ext=m4a]/best[height<=1080][ext=mp4]
4k
bestvideo[height<=2160][ext=mp4]+bestaudio[ext=m4a]/best[height<=2160][ext=mp4]
8k
bestvideo[height<=4320][ext=mp4]+bestaudio[ext=m4a]/best[height<=4320][ext=mp4]
 
Thank you very much for your answers.

**Comment by marcopeocchi** [comment]
@arbdevml 

In order I can implement in the immediate:

- [ ] Write comment of the video (3)
- [ ] Add Nginx reverse proxy config example (9)
- [ ] Videos subtitles (6)
- [ ] CRON based download (8)

Playlist / Channel support requires a *little more* work than the expected 🤠 
Authelia support: I have to study its architecture first since it's the first time that i hear about it.

Could you please explain more about nr. 2?

P.S. sorry for the late response, as I wrote on another issue, I've been a little overworked.

**Comment by arbdevml** [comment]
2. Add embed-thumbnail to the video file,
--embed-thumbnail Embed thumbnail in the video as cover art.
Thumbnail - video preview.

11. Create a download template. Save the download argument to reuse.
Example of a template:
yt-dlp -f \
'bestvideo[height<=1080][ext=mp4]+bestaudio[ext=m4a]/best[height<=1080][ext=mp4]' \
--merge-output-format mp4 \
--write-comments \
--write-description \
--write-info-json \
--mtime \
--continue \
--write-thumbnail \
--embed-thumbnail \
--embed-chapters \
--embed-metadata \
--batch-file \

-a, --batch-file FILE           File containing URLs to download ("-" for
                                stdin), one URL per line. Lines starting
                                with "#", ";" or "]" are considered as
                                comments and ignored
--date DATE                     Download only videos uploaded on this date.
                                The date can be "YYYYMMDD" or in the format 
                                [now|today|yesterday][-N[day|week|month|year]].
                                E.g. "--date today-2weeks" downloads only
                                videos uploaded on the same day two weeks ago
--datebefore DATE               Download only videos uploaded on or before
                                this date. The date formats accepted is the
                                same as --date
--dateafter DATE                Download only videos uploaded on or after
                                this date. The date formats accepted is the
                                same as --date
                                
Thank you for your reply. What do you think about implementing a download template?

**Comment by xonel** [comment]
> 2. Add embed-thumbnail to the video file,
>        --embed-thumbnail Embed thumbnail in the video as cover art.
>        Thumbnail - video preview.
> 
>     3. Create a download template. Save the download argument to reuse.
>        Example of a template:
>        yt-dlp -f 
>        'bestvideo[height<=1080][ext=mp4]+bestaudio[ext=m4a]/best[height<=1080][ext=mp4]' 
>        --merge-output-format mp4 
>        --write-comments 
>        --write-description 
>        --write-info-json 
>        --mtime 
>        --continue 
>        --write-thumbnail 
>        --embed-thumbnail 
>        --embed-chapters 
>        --embed-metadata 
>        --batch-file \
> 
> 
> -a, --batch-file FILE File containing URLs to download ("-" for stdin), one URL per line. Lines starting with "#", ";" or "]" are considered as comments and ignored --date DATE Download only videos uploaded on this date. The date can be "YYYYMMDD" or in the format [now|today|yesterday][-N[day|week|month|year]]. E.g. "--date today-2weeks" downloads only videos uploaded on the same day two weeks ago --datebefore DATE Download only videos uploaded on or before this date. The date formats accepted is the same as --date --dateafter DATE Download only videos uploaded on or after this date. The date formats accepted is the same as --date
> 
> Thank you for your reply. What do you think about implementing a download template?

**Comment by CrazyWolf13** [comment]
Yeah premade templates would be great, so I could save a template for video one for audio and then while entering the link just select one of my templates or enter custom...

**Comment by CrazyWolf13** [comment]
> Hello. First of all very big thank you for this project. What do you think about these:
> 
> 1. Add a date picker to download videos from the channel by specific date range,
>    --datebefore DATE               Download only videos uploaded on or before
>    this date. The date formats accepted is the
>    same as --date
>    --dateafter DATE                Download only videos uploaded on or after
>    this date.
> 2. Add embed-thumbnail to the video file,
>    --embed-thumbnail     Embed thumbnail in the video as cover art.
> 3. Write comment of the video,
>    --write-comments
> 4. Add comment viewer,
>    https://gist.github.com/pukkandan/ee737fec64822f2552caf3ca4cbf5db7
> 5. Write a video description of the video,
>    --write-description   Write video description to a .description file.
> 6. Download subtitles of the video,
>    Select the language and download the srt file of the subtitle.
> 7. Download the channel or playlist.
> 8. Automatically download more the new videos from channel or playlist,
>    (Can be implemented using CRON and API request).
> 9. Add Nginx reverse proxy config example.
> 10. Add example integration with Authelia.
> 11. Create a download template. Save the download argument to reuse.
> 12. Ability to download mp4, add drop-down list with video formats,
>     480p
>     bestvideo[height<=480][ext=mp4]+bestaudio[ext=m4a]/best[height<=480][ext=mp4]
>     720p
>     bestvideo[height<=720][ext=mp4]+bestaudio[ext=m4a]/best[height<=720][ext=mp4]
>     1080p
>     bestvideo[height<=1080][ext=mp4]+bestaudio[ext=m4a]/best[height<=1080][ext=mp4]
>     4k
>     bestvideo[height<=2160][ext=mp4]+bestaudio[ext=m4a]/best[height<=2160][ext=mp4]
>     8k
>     bestvideo[height<=4320][ext=mp4]+bestaudio[ext=m4a]/best[height<=4320][ext=mp4]
> 
> Thank you very much for your answers.



> @arbdevml
> 
> In order I can implement in the immediate:
> 
> * [ ]  Write comment of the video (3)
> * [ ]  Add Nginx reverse proxy config example (9)
> * [ ]  Videos subtitles (6)
> * [ ]  CRON based download (8)
> 
> Playlist / Channel support requires a _little more_ work than the expected 🤠 Authelia support: I have to study its architecture first since it's the first time that i hear about it.
> 
> Could you please explain more about nr. 2?
> 
> P.S. sorry for the late response, as I wrote on another issue, I've been a little overworked.

In addition to Authelia, what about Authentik ?
Authentik is more GUI based, supports a lot more methods of signing in, i guess it fits better
